File: Programmers/Lv.2/기능개발/42586.js
function solution(progresses, speeds) {
    const answer = [];
    
    let i = 0;
    while (i < progresses.length) {
        const addDays = parseInt(Math.ceil( (100 - progresses[i]) / speeds[i] ));
        
        for (let j = i; j < progresses.length; j++) {
            progresses[j] += (speeds[j] * addDays);
        }
        
        let deployCnt = 0;
        for (let k = i; k < progresses.length; k++) {
            if (progresses[k] < 100) break;
            deployCnt++;
        }
        answer.push(deployCnt);
        
        i += deployCnt;
    }
    
    return answer;
}

File: Programmers/Lv.2/기능개발/42586.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] answer = {};
        
        List<Integer> list = new ArrayList<>();
        int i = 0;
        
        while (i < progresses.length) {
            int addDays = (int) Math.ceil((double) (100 - progresses[i]) / speeds[i]);
            
            for (int j = i; j < progresses.length; j++) {
                progresses[j] += (speeds[j] * addDays);
            }
            
            int deployCnt = 0;
            for (int k = i; k < progresses.length; k++) {
                if (progresses[k] < 100) break;
                deployCnt++;
            }
            list.add(deployCnt);
            
            i += deployCnt;
        }
        
        answer = new int[list.size()];
        for (int n = 0; n < list.size(); n++) {
            answer[n] = list.get(n);
        }
        
        return answer;
    }
}
